### PR TITLE
{bio}[GCC/12.3.0] Salmon v1.10.3

### PR DESCRIPTION
**Pull request by pavelToman**
(created using `eb --new-pr`)
resolves https://github.com/vscentrum/vsc-software-stack/issues/384

### CODE REVIEW

**Comment by pavelToman** [comment]
@boegelbot please test @ generoso

**Comment by pavelToman** [comment]
@boegelbot please test @ jsc-zen3

**Comment by verdurin** [comment]
Test report by @verdurin
**SUCCESS**
Build succeeded for 5 out of 5 (1 easyconfigs in total)
easybuild-el8.cloud.in.bmrc.ox.ac.uk - Linux Rocky Linux 8.10, x86_64, Intel Xeon Processor (Skylake, IBRS), Python 3.6.8
See https://gist.github.com/verdurin/6624ae2e219973c42a8df2adf4ece891 for a full test report.

File: easybuild/easyconfigs/s/Salmon/Salmon-1.10.3-GCC-12.3.0.eb [hunk 1]
```diff
@@ -0,0 +1,59 @@
...
+homepage = 'https://github.com/COMBINE-lab/salmon'
+description = """Salmon is a wicked-fast program to produce a highly-accurate,
+ transcript-level quantification estimate from RNA-seq data."""
+
+toolchain = {'name': 'GCC', 'version': '12.3.0'}
+toolchainopts = {'pic': True}
+
+source_urls = ['https://github.com/COMBINE-lab/salmon/archive']
+sources = [{'download_filename': 'v%(version)s.tar.gz', 'filename': SOURCE_TAR_GZ}]
+checksums = ['a053fba63598efc4ade3684aa2c8e8e2294186927d4fcdf1041c36edc2aa0871']
```
Comment (verdurin) [inline easybuild/easyconfigs/s/Salmon/Salmon-1.10.3-GCC-12.3.0.eb:30]: @pavelToman is there a specific reason why you've changed the way the source download is specified?

Comment (pavelToman) [reply easybuild/easyconfigs/s/Salmon/Salmon-1.10.3-GCC-12.3.0.eb:30]: If I got it @boegel ask me to make `source` like this: https://github.com/easybuilders/easybuild-easyconfigs/pull/20852#discussion_r1705629115

Comment (verdurin) [reply easybuild/easyconfigs/s/Salmon/Salmon-1.10.3-GCC-12.3.0.eb:30]: I don't disagree, just wasn't aware of the change in policy, if that's what it is.
@boegel?

Comment (smoors) [reply easybuild/easyconfigs/s/Salmon/Salmon-1.10.3-GCC-12.3.0.eb:30]: renaming is not needed here as there's only 1 source tarball (no confusion possible), but it doesn't hurt

**Review by smoors — APPROVED** [review]
lgtm

**Comment by smoors** [comment]
Going in, thanks @pavelToman!

**Pull request merged**